modules/apps/29-fee/keeper/escrow.go

```diff
@@ -85,31 +86,36 @@ func (k Keeper) DistributePacketFeesOnTimeout(ctx sdk.Context, timeoutRelayer sd
 		}
 
 		// refund receive fee for unused forward relaying
-		k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.RecvFee)
+		k.distributeFee(ctx, refundAddr, refundAddr, feeInEscrow.Fee.RecvFee)
 
 		// refund ack fee for unused reverse relaying
-		k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.AckFee)
+		k.distributeFee(ctx, refundAddr, refundAddr, feeInEscrow.Fee.AckFee)
 
 		// distribute fee for timeout relaying
-		k.distributeFee(ctx, timeoutRelayer, feeInEscrow.Fee.TimeoutFee)
+		k.distributeFee(ctx, timeoutRelayer, refundAddr, feeInEscrow.Fee.TimeoutFee)
 	}
 }
 
 // distributeFee will attempt to distribute the escrowed fee to the receiver address.
 // If the distribution fails for any reason (such as the receiving address being blocked),
 // the state changes will be discarded.
-func (k Keeper) distributeFee(ctx sdk.Context, receiver sdk.AccAddress, fee sdk.Coins) {
+func (k Keeper) distributeFee(ctx sdk.Context, receiver, refundAccAddress sdk.AccAddress, fee sdk.Coins) {
 	// cache context before trying to distribute fees
 	cacheCtx, writeFn := ctx.CacheContext()
 
 	err := k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, receiver, fee)
-	if err == nil {
-		// write the cache
-		writeFn()
-
-		// NOTE: The context returned by CacheContext() refers to a new EventManager, so it needs to explicitly set events to the original context.
-		ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
+	if err != nil && !bytes.Equal(receiver, refundAccAddress) {
+		err := k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, refundAccAddress, fee)
+		if err != nil {
+			return
```

Comment:
Is there anything we can do here in the case of both failing? Emit some events?

Reply:
What would be the goal for emitting the event? Events are mostly just used by relayers I guess. 

I think it would be worth logging something to `INFO` or `DEBUG` logs. Although I haven't seen SDK modules utilizing logging much.

Reply:
> What would be the goal for emitting the event?

Just some kind of signalling I guess, was just thinking aloud really. I think logging at `WARN` or `ERROR` here may be useful, but like you say, most sdk modules don't seem to do much logging. 
Feels a little wrong to just silently no-op, that's my only concern.

Reply:
Yeah, I know what you mean. I guess the problem is that even if we returned an error here we would simply dismiss it at the level above anyway, otherwise, we would err on `OnAcknowledgementPacket` which would throw away the ack for the ics20 transfer or w/e as well. 

I think we're limited in what we can do here so my vote would be some warn/err logs.

Reply:
Seems like a bug if we cannot refund to the original sender? Maybe we should panic

Reply:
Hmm, maybe we should... originally @AdityaSripal had suggested to just no-op in the issue.

Reply:
I guess the no-op is for the case that the other fees were distributed fine but this one errors. If we err or panic the other fees won't ever get distributed.